docs/misc/hello_world_feature_TT.md

```diff
@@ -235,8 +249,96 @@ Hello George!
 For building the new feature into a physical Trezor, refer to [embedded](core/build/embedded.md).
 
 ## Testing
-TO BE DONE
+It is always good to include some tests exercising the created functionality, so when we break it later, it will be noticed. Trezor model T supports both `unit tests` and `integration tests` (which are called `device tests`).
+
 ### Unit tests
 [docs](core/tests/index.md)
+
+Unit tests can verify individual (mostly helper) functions that have clearly defined inputs and outputs.
+
+They are stored under `core/tests` and can be run by `make test` in the `core` directory.
+
+To call a specific test (the one we are about to create), run `make test TESTOPTS=test_apps.misc.hello_world.py`
+
+#### **`core/tests/test_apps.misc.hello_world.py`**
+```python
+from common import *
+
+from trezor.messages import HelloWorldRequest
+from apps.misc.hello_world import _get_text_from_msg
+
+
+class TestHelloWorld(unittest.TestCase):
+    def test_get_text_from_msg(self):
+        msg = HelloWorldRequest(name="Satoshi", amount=2, show_display=False)
+        self.assertEqual(_get_text_from_msg(msg), "Hello Satoshi!\nHello Satoshi!\n")
+
+
+if __name__ == "__main__":
+    unittest.main()
+```
+
+Code above is using the `unittest` testing framework, however not directly from python's standard library. As these unit tests are run by `micropython`, which does not have `unittest` library, we had to create the functionality ourselves in `core/tests/unittest.py` - see `from common import *`.
+
+Current code checks one usage of `_get_text_from_msg`, the only deterministic helper function we use in our feature. One could create many test vectors trying different inputs and expecting different outputs.
+
 ### Device tests
 [docs](tests/device-tests.md)
+
+Device tests (our name for integration tests) should test the whole workflow from sending the first request into Trezor to Trezor sending the final response.
+
+`Trezorlib` is used extensively in these tests as a way to request something from Trezor and then assert the expected response (it actually uses the code we created in Part 3).
```

Comment:
when `codified`, you should use lowercase `trezorlib` even at start of sentence

Reply:
Is there something else worth changing? Otherwise, I could force-push rebase with this change

Reply:
please go ahead.

Reply:
Done in [8596289](https://github.com/trezor/trezor-firmware/pull/2148/commits/8596289b4eaa6debc7b31a39a21dec3413cada8c)